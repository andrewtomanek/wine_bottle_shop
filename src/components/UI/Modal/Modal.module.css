.Modal {
  grid-area: 1/1/1/1;
  z-index: 5;
  background-color: white;
  width: 80%;
  border: 1px solid #ccc;
  box-shadow: 1px 1px 1px black;
  padding: 0;
  box-sizing: border-box;
  transition: all 0.3s ease-out;
}

/* @media (min-width: 600px) {
    .Modal {
        width: 500px;
        left: calc(50% - 250px);
    }
} */
