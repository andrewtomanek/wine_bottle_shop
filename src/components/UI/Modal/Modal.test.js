import React from "react";
import { shallow } from "enzyme";
import Modal from "./Modal";

const props = {
  show: true,
  modalClosed: false,
};

const closedProps = {
  show: false,
  modalClosed: false,
};

describe("Shallow Render Modal component", () => {
  it("renders backdrop", () => {
    const wrapper = shallow(<Modal {...props} />);
    expect(wrapper.find("Fragment").text()).toContain("<backdrop />");
  });

  it("renders Modal", () => {
    const wrapper = shallow(<Modal {...props} />);
    expect(wrapper.find({ className: "Modal" })).toHaveLength(1);
  });

  it("shows opacity and transform", () => {
    const wrapper = shallow(<Modal {...props} />);
    expect(
      wrapper.find({
        style: {
          opacity: "1",
          transform: "translateY(0)",
        },
      })
    ).toHaveLength(1);
  });

  it("hides opacity and transform", () => {
    const wrapper = shallow(<Modal {...closedProps} />);
    expect(
      wrapper.find({
        style: {
          opacity: "0",
          transform: "translateY(-100vh)",
        },
      })
    ).toHaveLength(1);
  });

  it("Match Snapshot of Modal component", () => {
    const wrapper = shallow(<Modal {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
