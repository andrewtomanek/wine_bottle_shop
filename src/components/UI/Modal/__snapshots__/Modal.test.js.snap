// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Render Modal component Match Snapshot of Modal component 1`] = `
<Fragment>
  <backdrop
    clicked={false}
    show={true}
  />
  <div
    className="Modal"
    style={
      Object {
        "opacity": "1",
        "transform": "translateY(0)",
      }
    }
  />
</Fragment>
`;
